crates/tendermint/src/chain/id.rs

```diff
@@ -0,0 +1,147 @@
+//! Tendermint blockchain identifiers
+
+#[cfg(feature = "serializers")]
+use serde::{de::Error as DeError, Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    cmp::Ordering,
+    fmt::{self, Debug, Display},
+    hash::{Hash, Hasher},
+    str::{self, FromStr},
+};
+
+use error::Error;
+
+/// Maximum length of a `chain::Id` name
+pub const MAX_LENGTH: usize = 64;
```

Comment:
This seemed reasonable enough but if there's a better number let me know

Reply:
ChainID will eventually become the hash of the  `ChainDescription`: https://github.com/tendermint/tendermint/blob/develop/docs/architecture/adr-017-chain-versions.md#chainid

Reply:
For now, it's actually capped at 50 UTF-8 characters: https://github.com/tendermint/tendermint/blob/develop/types/genesis.go#L17